### DISCUSSION TITLE
Many components have a lot of room for improvement

### DISCUSSION THREAD

**Discussion by 18651440358** [tusen-ai/naive-ui#5012] [Ideas]
Dear author, I am very grateful for the UI library you developed, I think many components have a lot of room for improvement:
### 1.  “progress” component join **step**
<img width="441" alt="截屏2023-07-02 11 37 49" src="https://github.com/tusen-ai/naive-ui/assets/44701892/1b60fd08-03d9-4b4f-9290-868d633ac42d">

### 2. Added component "ResizeBox"
![iShot_2023-07-02_11 55 17](https://github.com/tusen-ai/naive-ui/assets/44701892/417262fb-1a70-4070-abee-3bb62c07a2aa)
![iShot_2023-07-02_12 07 39](https://github.com/tusen-ai/naive-ui/assets/44701892/7938bdf5-294c-4838-8c9b-fdcb27e72d28)

**Comment by jahnli** [comment]
`n-split`
